### ISSUE TITLE
False positive from Style/ArrayIntersect in 1.88.2

### ISSUE THREAD

**Issue by tomhughes** [rubocop/rubocop#15437]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Style/ArrayIntersect

### Steps to reproduce the problem

In 1.88.2 the Style/ArrayIntersect cop has started making this suggestion:

```
app/models/oauth2_application.rb:44:28: C: [Correctable] Style/ArrayIntersect: Use scopes.intersect?(Oauth::PRIVILEGED_SCOPES) instead of scopes.any? { |scope| Oauth::PRIVILEGED_SCOPES.include?(scope) }.
    errors.add(:scopes) if scopes.any? { |scope| Oauth::PRIVILEGED_SCOPES.include?(scope) }
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

but it's wrong as `scopes` is an instance of `Doorkeeper::OAuth::Scopes` which does not define an `intersect?` method.

I assume this is a regression caused by #15421.

### Expected behavior

No report is made in this case.

### Actual behavior

A report is made in this case.

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
1.88.2 (using Parser 3.3.11.1, Prism 1.9.0, rubocop-ast 1.50.0, analyzing as Ruby 3.3, running on ruby 4.0.5) [x86_64-linux]
  - rubocop-capybara 3.0.0
  - rubocop-factory_bot 2.28.0
  - rubocop-minitest 0.39.1
  - rubocop-performance 1.26.1
  - rubocop-rails 2.35.5
  - rubocop-rake 0.7.1
```

**Comment by koic** [comment]
Thanks for the report. However, this is the documented behavior of this unsafe cop rather than a bug.

`Style/ArrayIntersect` is marked as `Safe: false` precisely because it cannot guarantee that the receiver is actually an `Array`:

> This cop cannot guarantee that `array1` and `array2` are actually arrays since `intersect?` is defined only for arrays.

https://docs.rubocop.org/rubocop/cops_style.html#stylearrayintersect

The same class of report was previously made in #11412, where the receiver was an `ActiveRecord::Associations::CollectionProxy`. That was resolved by marking this cop as unsafe.

Also, this is not specific to PR #15421. The block form has been detected since RuboCop 1.66, so the following code has always registered the same offense:

```ruby
scopes.any? { |scope| Oauth::PRIVILEGED_SCOPES.member?(scope) }
```

PR #15421 only extended the existing matcher to accept `include?`, which is an alias of `member?` on `Array`.

If `intersect?` is not defined for your object, you can disable the cop for that line with `# rubocop:disable Style/ArrayIntersect`, run RuboCop with the `--safe` option, or disable the cop in your configuration.
